spec/unit/models/project_creator_spec.rb

```diff
@@ -71,4 +71,20 @@
       GoodData::Model::ProjectCreator.pick_correct_chunks(@data, update_preference: { cascade_drops: true, preserve_data: false, unmeetable_condition: true })
     end.to raise_error
   end
+
+  it 'should pick correct update chunks based on new parameter "allow_cascade_drops" in your preference' do
+    chunk = GoodData::Model::ProjectCreator.pick_correct_chunks(@data, update_preference: { allow_cascade_drops: false })
+    expect(chunk).to eq [@chunk_a]
+  end
+
+  it 'should pick correct update chunks based on new parameter "keep_data" in your preference' do
+    chunk = GoodData::Model::ProjectCreator.pick_correct_chunks(@data, update_preference: { keep_data: false })
+    expect(chunk).to eq [@chunk_a, @chunk_b]
+  end
+
+  it 'should raise error when mixing new parameters with the old ones in your preference' do
+    expect do
+      GoodData::Model::ProjectCreator.pick_correct_chunks(@data, update_preference: { allow_cascade_drops: true, cascade_drops: false })
+    end.to raise_error
```

Comment:
Please check that _the right_ error was raised by passing a parameter to `raise_error`.